cores/esp32/HWCDC.cpp

```diff
@@ -308,15 +452,28 @@ void HWCDC::flush(void)
     if(xSemaphoreTake(tx_lock, tx_timeout_ms / portTICK_PERIOD_MS) != pdPASS){
         return;
     }
-    UBaseType_t uxItemsWaiting = 0;
-    vRingbufferGetInfo(tx_ring_buf, NULL, NULL, NULL, NULL, &uxItemsWaiting);
-    if(uxItemsWaiting){
-        // Now trigger the ISR to read data from the ring buffer.
-        usb_serial_jtag_ll_ena_intr_mask(USB_SERIAL_JTAG_INTR_SERIAL_IN_EMPTY);
-    }
-    while(uxItemsWaiting){
-        delay(5);
+    if(!isCDC_Connected()) {
+        flushTXBuffer(NULL, 0);
+    } else {
+        UBaseType_t uxItemsWaiting = 0;
         vRingbufferGetInfo(tx_ring_buf, NULL, NULL, NULL, NULL, &uxItemsWaiting);
+        if(uxItemsWaiting){
+            // Now trigger the ISR to read data from the ring buffer.
+            usb_serial_jtag_ll_txfifo_flush();
+            if(connected) usb_serial_jtag_ll_ena_intr_mask(USB_SERIAL_JTAG_INTR_SERIAL_IN_EMPTY);
+        }
+        uint32_t tries = tx_timeout_ms; // waits 1ms per ISR sending data attempt, in case CDC is unplugged
+        while(connected && tries && uxItemsWaiting){
+            delay(1);
```

Comment:
delay(1) might not always be sufficient to allow other tasks to run. I suggest you use 2 here

Reply:
Yes, this is just to wait for the ISR (HOST to request all data).